vulture/core.py

```diff
@@ -518,6 +487,58 @@ def visit_Call(self, node):
             if isinstance(attr_name_arg, ast.Str):
                 self.used_names.add(attr_name_arg.s)
 
+        # Parse variable names in new format strings:
+        # '{my_var}'.format(**locals())
+        if (
+            isinstance(node.func, ast.Attribute)
+            and isinstance(node.func.value, ast.Str)
+            and node.func.attr == "format"
+        ):
+            if any(
+                kw.arg is None and self._is_locals_call(kw.value)
+                for kw in node.keywords
+            ):
+                self._handle_new_format_string(node.func.value.s)
+
+    def _handle_new_format_string(self, s):
+        def is_identifier(name):
+            return bool(re.match(r"[a-zA-Z_][a-zA-Z0-9_]*", name))
+
+        parser = string.Formatter()
+        try:
+            names = [name for _, name, _, _ in parser.parse(s) if name]
+        except ValueError:
+            # Invalid format string.
+            names = []
+
+        for field_name in names:
+            # Remove brackets and their contents: "a[0][b].c[d].e" -> "a.c.e",
+            # then split the resulting string: "a.b.c" -> ["a", "b", "c"]
+            vars = re.sub(r"\[\w*\]", "", field_name).split(".")
+            for var in vars:
+                if is_identifier(var):
+                    self.used_names.add(var)
+
+    def visit_BinOp(self, node):
+        """
+        Parse variable names in old format strings:
+
+        '%(my_var)s' % locals()
+        """
+        if isinstance(node.left, ast.Str) and self._is_locals_call(node.right):
```

Comment:
Let's test that the BinOp operator is "%".

Reply:
nice catch, updated the tests as well to cover this omission